cl/search/api_serializers.py

```diff
@@ -306,11 +321,20 @@ def create(self, validated_data):
         "boolean": serializers.BooleanField,
         "text": serializers.CharField,
         "keyword": serializers.CharField,
-        "date": serializers.DateTimeField,
+        "date": DateOrDateTimeField,
```

Comment:
yeah, this is not required anymore, as we talked about, for now we'll go with the current Datetime format for date fields. When working on the new Search API endpoints we'll consider using naive datetimes for values that doesn't have a timezone.

I didn't require to use `midnight_pt` since by default `serializers.DateTimeField` adds the system timezone in this case (PT).

So date fields preserver the same format:

```
 {
            "absolute_url": "/audio/10/nelson-plc-v-gomez-bell-and-moreno/",
            "caseName": "Nelson PLC v. Gomez, Bell and Moreno",
            "court": "Thirteenth circuit of the Eruptanyom",
            "court_citation_string": "",
            "court_exact": "tuubd",
            "court_id": "tuubd",
            "dateArgued": "1971-02-07T00:00:00-08:00",
            "dateReargued": null,
            "dateReargumentDenied": null,
            "docketNumber": "4:05-ms-35426",
            "docket_id": 10,
            "download_url": "http://rose.net/",
            "duration": null,
            "file_size_mp3": 0,
            "id": 10,
            "judge": null,
            "local_path": "dat/1971/02/07/example_4.dat",
            "pacer_case_id": "313861",
            "panel_ids": [],
            "sha1": "4d6649a0ee4312a86c12147d0a275b59751f892f",
            "snippet": null,
            "source": "M",
            "timestamp": "2023-08-31T19:23:29.460227-07:00"
        }
```

Reply:
Ok, this changed again. As we talked about, this also impact on webhooks. 

On webhooks we'll start showing date values as dates instead of transforming them to datetime fields (e.g: dateArgued, dateFiled, etc).

On the API, current version will continue showing date fields with datetime values using the system timezone (PT).

datetime fields will show datetime values e.g: timestamp.

So I added an improved version of `DateOrDateTimeField`

So according to the field value received it ill render a `date` or `datetime`.

The difference we have between the API and webhooks is that when we retrieve results from Elasticsearch, we get a naive `datetime` value for all date fields. However, for sending webhooks, we don’t perform an additional query to the OA index in ES to fetch the document. Instead, we save the document in the database, prepare it (extracting the fields we usually send to ES when indexing the document), and then index it. If an alert is triggered, we use this prepared version to send alerts. This version contains fields as they are in the database. As a result, date fields are in `date` format, not in `datetime` format as they are when retrieved from ES.